### ISSUE TITLE
host the extension on Chrome Web Store

### ISSUE THREAD

**Issue by caver456** [ncssar/gpsio#13]
The extension hasn't been available on the Chrome Web Store for some time - back when it was hosted by caltopo/Matt.  Looks like it needs to be brought up to Chrome's latest standards.

This is a good opportunity to revisit Chrome's extension docs from the top, and do the work to make the extension fully compliant: https://developer.chrome.com/docs/extensions/

**Comment by caver456** [comment]
- [ ] create a new shared account just for Chrome Web Store (and possibly Firefox extension hosting as needed)
- [ ] upgrade to MV3 (Manifest V3)
- [ ] decide whether the extension ID should be hardcoded (using the 'key' element in manifest.json) - see #18 
- [ ] test and pack the extension locally (note, the native messaging host, registry edit, Python (or executable) install, GPSBabel install, Garmin drivers install (including .NET install) are all separate processes, ideally rolled into a single installer
- [ ] submit for Chrome Web Store approval
- [ ] modify and resubmit as necessary to get it hosted

**Comment by caver456** [comment]
Google does not allow an upload with 'key' in the manifest.  So it looks like it may not be possible to keep the same extension ID as earlier versions of the extension, since that extension is no longer hosted therefore its ownership cannot be transferred.  Not the end of the world, since the eventual installer will install this extension anyway.

**Comment by caver456** [comment]
Google does not like 'host permissions' - in this case, apparently the 'host permissions' are the 'matches' element values in the content_scripts element in manifest.json.  Explore whether these entries are really needed - life will be simpler if we get rid of them.  The MV3 migration docs are a bit unclear about this, but, just try getting rid of them.

**Comment by caver456** [comment]
submitted for review Nov 1 2021.

**Comment by caver456** [comment]
CWS rejected the extension:

-----

Hi there,

We regret to inform you that the most recent submission of your item was rejected. Please find the details below.


Item name: gpsio

Item ID: cbpembjdolhcjepjgdkcflipfojbjall

Violation(s):

Use of Permissions:

Violation reference ID: Purple Potassium
Violation: The following permission(s) need not be requested for the methods/properties implemented by the item:
nativeMessaging
How to rectify: Remove the above permission(s). The properties used by the items will continue to function even without requesting these permissions.
Relevant section of the program policy:
Request access to the narrowest permissions necessary to implement your Product's features or services. Don't attempt to "future proof" your Product by requesting a permission that might benefit services or features that have not yet been implemented. (learn more)


Do you believe this was a mistake?

You can use this Support Page to learn more about this violation and get in touch with us.


We value developer contributions to the Chrome Web Store, and look forward to helping you bring your item into compliance with our policies.

Thanks,

Chrome Web Store Developer Support

-----

I followed the links which let you file a detailed appeal; here's the wording I sent:

background.js (the service worker) uses chrome.runtime.connectNative (line 6 - the first line of function 'fn') - https://developer.chrome.com/docs/apps/nativeMessaging/#native-messaging-client clarifies that this requires the nativeMessaging permission.

-----

got this confirmation email:

Your Chrome Web Store Support ticket has been created [7-0977000031724]

**Comment by caver456** [comment]
The appeal was successful.  Added a note in the nativeMessaging justification field, referencing the ticket#, and resubmitted.

-----
Hello Developer,

Thank you for reaching out to us in detail with evidence. We sincerely apologize for the inconvenience caused to you in this matter. Upon a subsequent review, we found that your item with ID: “cbpembjdolhcjepjgdkcflipfojbjall” to be compliant with all our policies. Unfortunately, we cannot approve a submission that has been rejected. Hence, we request you to re-submit your extension on the developer dashboard.

We value your contributions to the Chrome Web Store and look forward to working with you.

Thanks,


Chrome Web Store Developer Support
